_posts/2023-06-09-flat-object.md

```diff
@@ -0,0 +1,365 @@
+---
+layout: post
+title:  "Use flat object in OpenSearch"
+authors:
+- mingshl
+- kolchfa
+date: 2023-06-09
+categories:
+ - technical-post
+meta_keywords: 
+meta_description: 
+
+excerpt: OpenSearch 2.7 introduced the new `flat_object` field type. This field type is useful for objects with a large number of fields or when you are not familiar with the field names in your documents. The `flat_object` field type treats the entire JSON object as a string. Subfields within the JSON object are accessible using the flat object field name or the standard dot path notation, but they are not indexed for fast lookup. In this post, we explore how flat object simplifies mapping data structures and enhances the search experience in OpenSearch. 
+---
+
+OpenSearch 2.7 introduced the new `flat_object` field type. This field type is useful for objects with a large number of fields or when you are not familiar with the field names in your documents. The `flat_object` field type treats the entire JSON object as a string. Subfields within the JSON object are accessible using the flat object field name or the standard dot path notation, but they are not indexed for fast lookup. In this post, we explore how flat object simplifies mapping data structures and enhances the search experience in OpenSearch. 
+
+## Dynamic mapping
+
+In OpenSearch, a _mapping_ defines the structure of your data. It specifies field names, types, and indexing and analysis settings, ensuring that your data is organized and interpreted correctly. If you don't specify a custom mapping, OpenSearch infers the structure of your document automatically when you upload the document. This process is called _dynamic mapping_, where OpenSearch detects the document data structure and generates the corresponding mapping file. 
```

Comment:
Shall we also link the mapping [doc](https://opensearch.org/docs/2.8/field-types/index/)?